Sources/MapboxCoreNavigation/PassiveLocationDataSource.swift

```diff
@@ -49,13 +53,43 @@ open class PassiveLocationDataSource: NSObject {
      The location data source’s delegate.
      */
     public weak var delegate: PassiveLocationDataSourceDelegate?
+
+    /**
+     Delegate for Electronic Horizon updates.
+     */
+    public weak var electronicHorizonDelegate: EHorizonDelegate? {
+        didSet {
+            if delegate != nil {
```

Comment:
Did you mean to check `electronicHorizonDelegate
 != nil` here? Otherwise I don't understand the relation with `delegate`.

Also, it is technically possible to have `PassiveLocationDataSource` and `RouteController` active at the same time and thus they'll compete for being the EH observer. Since both implementations just forward the calls from `NN.Navigator` which is `shared`, maybe it is logical to also move this delegate setting to `Navigator`?

Reply:
I ended up removing this virtual delegate in favor of a notification system.